### PR TITLE
Fix golangci-lint violations

### PR DESCRIPTION
**Pull request by anderseknert**
Not sure if these are from a local upgrade or we're not running this as we think we are in CI. Some rules here are definitely new, but not all of them. I guess existing linters could have seen fixes as well. Will look into it.

### CODE REVIEW

File: cmd/languageserver.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ func init() {
 		Short: "Run the Regal Language Server",
 		Long:  `Start the Regal Language Server and listen on stdin/stdout for client editor messages.`,
 
-		RunE: wrapProfiling(func(args []string) error {
+		RunE: wrapProfiling(func(_args []string) error {
```
Comment (srenatus) [inline cmd/languageserver.go:22]: I'd just do `func([]string)` in this and the similar cases...

Comment (anderseknert) [reply cmd/languageserver.go:22]: Maybe it's obvious enough in this case? But I did that first and it just looked like something I'd wonder about 3 months later.

File: cmd/new.go [hunk 2]
```diff
@@ -65,31 +65,31 @@ Example:
 
 regal new rule --type custom --category naming --name camel-case`,
 
-		PreRunE: func(cmd *cobra.Command, args []string) error {
+		PreRunE: func(_ *cobra.Command, _args []string) error {
```
Comment (srenatus) [inline cmd/new.go:68]:
```suggestion
		PreRunE: func(*cobra.Command, []string) error {
```

**Review by srenatus — APPROVED** [review]
LGTM. I'd kill the _arg names, but if you think they're helpful, fine with me 🙃

File: cmd/languageserver.go (hunk 1 above)
Comment (anderseknert) [reply cmd/languageserver.go]: Alright, I caved in to your demands :)

**Pull request merged**